src/frontend/src/tests/lib/utils/format.utils.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { ZERO } from '$lib/constants/app.constants';
+import { formatBigNumberish } from '$lib/utils/format.utils';
+import { describe, expect } from 'vitest';
+
+describe('formatBigNumberish', () => {
+	it('formats bigint correctly when the value needs to be rounded', () => {
+		const value = 100000009000099049545454n;
+		const result = 10000000.900009904;
```

Comment:
I have to say that mathematically, this seems wrong...  that number, when rounded, should be `10000000.900009905`, no?

Reply:
I'm looking for explanation why `100000009000099049545454n` results in `10000000.900009904` (not rounded), and `100000009000099059545454n` -> `10000000.900009906` (last 5 becomes 6). It's the same result no matter whether I do the div operation with numbers or BigNumbers. Any ideas?

Reply:
That is pretty strange... what if you use `formatUnits`?

Reply:
Yes, it's the same no matter what type or util is used to make the calculations. I think we are fine as long as both values we use for comparisons (e.g. `userAmount` and `sendBalance`) are formatted in the same way.